Shared/alerts/AlertConditionsController.cpp

```diff
@@ -708,154 +705,115 @@ void AlertConditionsController::onLayersChanged()
  */
 void AlertConditionsController::onMouseClicked(QMouseEvent &event)
 {
+  // abort if the tool is not the active tool for some unexplained reason
   if (!isActive())
     return;
 
+  // only respond to left click
   if (event.button() != Qt::MouseButton::LeftButton)
     return;
 
+  // abort if the tool is somehow not in pick mode
   if (!m_pickMode)
     return;
 
-  if (m_identifyLayersWatcher.isValid() && !m_identifyLayersWatcher.isDone())
-    return;
-
-  if (m_identifyGraphicsWatcher.isValid() && !m_identifyGraphicsWatcher.isDone())
-    return;
-
+  // get a pointer to the geoView for calling the identify operations
   GeoView* geoView = ToolResourceProvider::instance()->geoView();
   if (!geoView)
     return;
 
-  m_identifyLayersWatcher = geoView->identifyLayers(event.position().x(), event.position().y(), m_tolerance, false);
-  m_identifyGraphicsWatcher = geoView->identifyGraphicsOverlays(event.position().x(), event.position().y(), m_tolerance, false);
-
-  event.accept();
-}
-
-/*!
-  \brief internal
-
-  Handle the result of an identify layers task.
- */
-void AlertConditionsController::onIdentifyLayersCompleted(const QUuid& taskId, QList<IdentifyLayerResult*> identifyResults)
-{
-  if (taskId != m_identifyLayersWatcher.taskId())
-    return;
-
-  LayerResultsManager resultsManager(identifyResults);
-
-  if (!isActive())
-    return;
-
-  m_identifyLayersWatcher = TaskWatcher();
+  // dispatch identify calls to the graphics layers and operational layers of the GeoView
+  // TODO: can we do something here to search only the layer in the combobox? right now, the user must toggle visibility
```

Comment:
Confirmed in the previous code, this will only ever return the first feature that is encountered regardless of the layer the user has selected in the dropdown.

Will create a new issue for this correction.